### ISSUE TITLE
Perform checksum check when creating the contract instance with existing address

### ISSUE THREAD

**Issue by nikita-fuchs** [aeternity/aepp-sdk-js#737]
Currently, the SDK lets me create a contract instance with any address, whether it's an invalid or simply a wrong one ( wrong = the expected contract does not exist there) 

e.g. 
```
myContract = await this.Chain.getContractInstance(this.code, {contractAddress: someBogusAddress});
```
Only after that, when calling a function, there is a check performed (at least it looks so), whether there is actually anything suitable behind the address, instead of performing checks when creating the instance right away.

**But :** we need to do this in intelligent way. 

**Throw** an error if:
- Always when there is absolutely no contract code at the entered address.

**Don't throw** an error if:
- Provide a flag in the `{options}` during [contract instance creation](https://github.com/aeternity/aepp-sdk-js/blob/8cd3287937284620602b2dddae837e2ecbeddf7d/docs/api/contract/aci.md#getcontractinstancesource-options--contractinstance-) to allow the instance to be created if the current code just _diverges_ from the one found on chain. 

That leads us to the following: The SDK's contract instances will never point to an empty address. At your own risk though, you can still interact with an "unknown" contract if you have at least a part of the contract ("an interface"). This might be useful aepps having some token logic and additional logic. If you know the token standard you can still use the token functionality of the contract without knowing the whole contract code.

**Comment by noandrea** [comment]
this was reverted due to an issue with the bytecode check, that will be fixed once #784 is done

**Comment by nikita-fuchs** [comment]
think we even have an API endpoint for this now
